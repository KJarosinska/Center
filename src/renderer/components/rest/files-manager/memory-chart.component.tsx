import React from "react"

import FunctionComponent from "Renderer/types/function-component.interface"

const MemoryChart: FunctionComponent = props => {
  console.log(props)
  return <div>Memorychart</div>
}

export default MemoryChart
