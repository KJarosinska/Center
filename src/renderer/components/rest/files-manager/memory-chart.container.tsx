import { connect } from "react-redux"
import MemoryChart from "Renderer/components/rest/files-manager/memory-chart.component"
import { select } from "Renderer/store"

const mapStateToProps = select(({ filesManager }) => ({
  data: filesManager.withConvertedBytes,
}))

export default connect(mapStateToProps, null)(MemoryChart)
