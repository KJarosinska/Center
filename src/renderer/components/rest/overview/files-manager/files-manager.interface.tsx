export interface FilesManagerProps {
  onFilesOpen: () => void
  usedSpace: number
  maxSpace?: number
  unit?: "B" | "KB" | "MB" | "GB"
}
