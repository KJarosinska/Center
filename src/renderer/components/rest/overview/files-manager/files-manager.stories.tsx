import React from "react"
import FilesManager from "Renderer/components/rest/overview/files-manager/files-manager.component"
import styled from "styled-components"
import Text, {
  TextDisplayStyle,
} from "Renderer/components/core/text/text.component"
import { action } from "@storybook/addon-actions"
import { Story, Meta } from "@storybook/react/types-6-0"
import { MemoryRouter } from "react-router"

const Part = styled.div`
  padding: 2rem;
  > p {
    margin-bottom: 2rem;
  }
`

export const DifferentSpaceUsed = () => (
  <>
    <Part>
      <Text displayStyle={TextDisplayStyle.SmallText}>No space used</Text>
      <FilesManager usedSpace={0} onFilesOpen={action("open files manager")} />
    </Part>
    <Part>
      <Text displayStyle={TextDisplayStyle.SmallText}>Custom parameters</Text>
      <FilesManager
        usedSpace={29}
        maxSpace={512}
        onFilesOpen={action("open files manager")}
      />
    </Part>
    <Part>
      <Text displayStyle={TextDisplayStyle.SmallText}>Half space used</Text>
      <FilesManager
        usedSpace={7.99}
        onFilesOpen={action("open files manager")}
      />
    </Part>
    <Part>
      <Text displayStyle={TextDisplayStyle.SmallText}>Full space used</Text>
      <FilesManager usedSpace={16} onFilesOpen={action("open files manager")} />
    </Part>
  </>
)

export default {
  title: "Overview/FilesManager",
  component: DifferentSpaceUsed,
  decorators: [
    (Story: Story) => (
      <div style={{ maxWidth: "63rem" }}>
        <MemoryRouter initialEntries={["/phone"]}>
          <Story />
        </MemoryRouter>
      </div>
    ),
  ],
} as Meta
