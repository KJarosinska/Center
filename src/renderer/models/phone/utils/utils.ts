import Faker from "faker"

// TODO: remove before production
export const generateFakeData = (numberOfContacts: number) => {
  return Array(numberOfContacts)
    .fill(0)
    .map(_ => ({
      id: Faker.random.uuid(),
      firstName: Faker.name.firstName(),
      lastName: Faker.name.lastName(),
      phoneNumber: Faker.phone.phoneNumber(),
      favourite: Faker.random.boolean(),
    }))
}

export const generateSortedStructure = (fakeState: any) => {
  const alphabet = new Array(26)
    .fill(1)
    .map((_, i) => String.fromCharCode(65 + i))

  const sortedContactList = fakeState.sort((a: any, b: any) =>
    a.firstName.localeCompare(b.firstName)
  )

  const generateFakeStructure = () => {
    const fakeStructure = []

    for (const letter of alphabet) {
      fakeStructure.push({
        category: letter,
        contacts: [],
      })
    }

    fakeStructure.unshift({
      category: "Favs",
      contacts: [],
    })

    return fakeStructure
  }

  const placeContactsInStructure = () => {
    const fakeStructure: any = generateFakeStructure()
    fakeStructure.forEach((fakeContact: any) => {
      const contactLetter = fakeContact.category
      const contactList: string[] = fakeContact.contacts
      sortedContactList.forEach((sortedContact: any) => {
        if (
          sortedContact.firstName.charAt(0) === contactLetter &&
          !sortedContact.favourite
        ) {
          contactList.push(sortedContact)
        }
      })
    })

    sortedContactList.forEach((contacts: any) => {
      if (contacts.favourite) {
        fakeStructure[0].contacts.push(contacts)
      }
    })

    return fakeStructure
  }

  return placeContactsInStructure()
}

export const sanitizeContacts = (structure: any) => {
  return structure.filter((el: any) => el.contacts.length > 0)
}

export const filterContacts = (contacts: any, substring: string) => {
  return contacts.map((contactsByLetter: any) => ({
    ...contactsByLetter,
    contacts: contactsByLetter.contacts.filter((contact: any) => {
      const filterableFields = Object.values(contact).filter(value => {
        return typeof value === "string"
      })
      return filterableFields.some((value: any) =>
        value.toLowerCase().includes(substring.toLowerCase())
      )
    }),
  }))
}
