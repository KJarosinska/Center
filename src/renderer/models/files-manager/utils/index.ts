import { FilesManagerState } from "Renderer/models/files-manager/files-manager.interface"

export const prepareDataForStackedBarChart = (data: FilesManagerState) => {
  return data.memoryData.map(el => {
    const { filesType, filesAmount, occupiedMemory, ...rest } = el
    return {
      value: occupiedMemory,
      ...rest,
    }
  })
}
