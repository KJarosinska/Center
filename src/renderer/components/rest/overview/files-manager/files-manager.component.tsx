import React from "react"
import FunctionComponent from "Renderer/types/function-component.interface"
import { FilesManagerProps } from "Renderer/components/rest/overview/files-manager/files-manager.interface"
import Card, {
  CardAction,
  CardActionButton,
  CardText,
} from "Renderer/components/rest/overview/card.elements"
import StackedBarChart, {
  DisplayStyle,
} from "Renderer/components/core/stacked-bar-chart/stacked-bar-chart.component"
import styled from "styled-components"
import { intl } from "Renderer/utils/intl"
import Text, {
  TextDisplayStyle,
} from "Renderer/components/core/text/text.component"
import { FormattedMessage } from "react-intl"
import {
  backgroundColor,
  fontWeight,
  letterSpacing,
  textColor,
} from "Renderer/styles/theming/theme-getters"
import FolderIcon from "Renderer/svg/files-manager.svg"

const TextInfo = styled(CardText)`
  > p {
    margin-top: 1.2rem;
    margin-bottom: 0.8rem;
    letter-spacing: ${letterSpacing("small")}rem;
    color: ${textColor("placeholder")};
  }
`

const SpaceData = styled.div`
  display: flex;
  align-items: baseline;

  span {
    margin-left: 0.8rem;
    font-weight: ${fontWeight("light")};
    text-transform: none;
  }
`

const BarChart = styled(StackedBarChart)`
  width: 12rem;
`

const Button = styled(CardActionButton)`
  svg * {
    fill: ${backgroundColor("light")};
  }
`

const FilesManager: FunctionComponent<FilesManagerProps> = ({
  className,
  usedSpace,
  maxSpace = 16,
  unit = "GB",
  onFilesOpen,
}) => {
  const stackedBarData = [
    {
      value: usedSpace,
      color: "#6d9bbc",
    },
    {
      value: maxSpace - usedSpace,
      color: "#f4f5f6",
    },
  ]

  const buttonLabel = intl.formatMessage({
    id: "view.name.overview.filesManager.openFilesManager",
  })

  return (
    <Card className={className}>
      <TextInfo>
        <SpaceData>
          <Text displayStyle={TextDisplayStyle.SecondaryBoldHeading}>
            {usedSpace}
          </Text>
          <Text element={"span"} displayStyle={TextDisplayStyle.SmallText}>
            /{maxSpace} {unit}
          </Text>
        </SpaceData>
        <Text displayStyle={TextDisplayStyle.SmallFadedText}>
          <FormattedMessage id="view.name.overview.filesManager.usedSpace" />
        </Text>
        <BarChart
          chartData={stackedBarData}
          displayStyle={DisplayStyle.Simple}
        />
      </TextInfo>
      <CardAction filled>
        <Button
          active
          label={buttonLabel}
          Icon={FolderIcon}
          onClick={onFilesOpen}
        />
      </CardAction>
    </Card>
  )
}

export default FilesManager
