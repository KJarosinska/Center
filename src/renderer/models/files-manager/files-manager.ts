import { Slicer } from "@rematch/select"
import { URL_MAIN } from "Renderer/constants/urls"
import { FilesManagerState } from "Renderer/models/files-manager/files-manager.interface"
import { prepareDataForStackedBarChart } from "Renderer/models/files-manager/utils"
import FilesManager from "Renderer/svg/files-manager.svg"
import Mudita from "Renderer/svg/mudita.svg"
import Music from "Renderer/svg/music.svg"
import VoiceRecorder from "Renderer/svg/voice-recorder.svg"

const initialStateValue: FilesManagerState = {
  memoryData: [
    {
      filesType: "Music",
      occupiedMemory: 5848538500,
      filesAmount: 15,
      color: "#6D9BBC",
      icon: Music,
      url: URL_MAIN.music,
    },
    {
      filesType: "Voice Recorder",
      occupiedMemory: 584850000,
      filesAmount: 3,
      color: "#AEBEC9",
      icon: VoiceRecorder,
      url: "/tools/voice-recorder",
    },
    {
      filesType: "Storage",
      occupiedMemory: 984443800,
      filesAmount: 85,
      color: "#E3F3FF",
      icon: FilesManager,
    },
    {
      filesType: "Free",
      occupiedMemory: 5848538522,
      filesAmount: 44,
      color: "#E9E9E9",
      icon: Mudita,
    },
  ],
}

export default {
  state: initialStateValue,
  selectors: (slice: Slicer<typeof initialStateValue>) => ({
    stackedBarChart() {
      return slice(state => {
        return prepareDataForStackedBarChart(state)
      })
    },
    memoryChart() {
      return slice(state => {
        return state.memoryData
      })
    },
  }),
}
