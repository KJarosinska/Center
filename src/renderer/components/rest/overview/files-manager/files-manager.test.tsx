import "@testing-library/jest-dom/extend-expect"
import React from "react"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"
import FilesManager from "Renderer/components/rest/overview/files-manager/files-manager.component"
import { FilesManagerProps } from "Renderer/components/rest/overview/files-manager/files-manager.interface"
import { noop } from "Renderer/utils/noop"
import { fireEvent } from "@testing-library/dom"
import { wait } from "@testing-library/react"

const renderFilesManager = ({
  onFilesOpen = noop,
  usedSpace = 0,
  ...props
}: Partial<FilesManagerProps> = {}) => {
  const outcome = renderWithThemeAndIntl(
    <FilesManager onFilesOpen={onFilesOpen} usedSpace={usedSpace} {...props} />
  )
  return {
    ...outcome,
  }
}

test("matches snapshot", () => {
  const { container } = renderFilesManager()
  expect(container).toMatchSnapshot()
})

test("renders space info properly", () => {
  const { getByText } = renderFilesManager({
    usedSpace: 33.5,
    maxSpace: 512,
    unit: "MB",
  })
  expect(getByText("33.5")).toBeInTheDocument()
  expect(getByText("/512 MB")).toBeInTheDocument()
})

test("triggers opening files manager after button click", async () => {
  const onFilesOpen = jest.fn()

  const { getByRole } = renderFilesManager({
    onFilesOpen,
  })

  fireEvent.click(getByRole("button"))

  await wait(() => {
    expect(onFilesOpen).toHaveBeenCalled()
  })
})
