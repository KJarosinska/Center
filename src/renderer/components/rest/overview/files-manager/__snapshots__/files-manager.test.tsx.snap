// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <section
    class="sc-dnqmqq cNtIGj"
  >
    <div
      class="sc-iwsKbI sc-kgoBCf HTlwO"
    >
      <div
        class="sc-kGXeez gSiFpA"
      >
        <h2
          class="sc-bdVaJa iSLnxf"
        >
          0
        </h2>
        <span
          class="sc-bdVaJa ghJYji"
        >
          /
          16
           
          GB
        </span>
      </div>
      <p
        class="sc-bdVaJa bAPoXv"
      >
        Used space
      </p>
      <div
        class="sc-VigVT cQowiX sc-kpOJdX gFJMka"
      >
        <div
          class="sc-jTzLTM brnJtJ"
        >
          <div
            class="sc-fjdhpX hzFurV"
            color="#6d9bbc"
            data-testid="bar"
          />
          <div
            class="sc-fjdhpX dQWmuK"
            color="#f4f5f6"
            data-testid="bar"
          />
        </div>
      </div>
    </div>
    <section
      class="sc-gzVnrw jKdxIN sc-gZMcBi FnzbX"
    >
      <button
        class="sc-ifAKCX fBjzQi sc-htoDjs sc-gqjmRU sc-dxgOiQ kCGqUM"
        size="2"
        type="button"
      >
        <test-file-stub
          classname="sc-EHOje cmLHsF"
        />
        <p
          class="sc-bdVaJa ghJYji sc-bZQynM flUvJk"
        >
          Open files manager
        </p>
      </button>
    </section>
  </section>
</div>
`;
